benchmarks/bench_merge.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Igor Kushnir <igorkuo@gmail.com>.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#include <cstdlib>
+#include <algorithm>
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <gfx/timsort.hpp>
+#include "benchmarker.hpp"
+
+template <typename value_t>
+struct Bench {
+    void operator()(const std::vector<value_t> &source) const {
+        const int size = static_cast<int>(source.size());
+        const std::vector<int> middle_positions = {
+            0, 1, 2, 5, 100, size/100, size/20, size/5, size/3, size/2, 3*size/4,
+            6*size/7, 24*size/25, 90*size/91, size-85, size-8, size-2, size-1, size
+        };
+
+        int result_count = 0;
+        Result result_sum = {};
+
+        std::cerr << "middle\\algorithm:\tstd::inplace_merge\ttimmerge" << std::endl;
+        constexpr int width = 10;
+        constexpr const char* padding1 = "       \t";
+        constexpr const char* padding2 = "        \t";
+
+        std::vector<value_t> a(source);
+        for (auto middle : middle_positions) {
+            if (middle < 0 || middle > size) {
+                continue;
+            }
+
+            std::copy(source.begin(), source.end(), a.begin());
```

Comment:
You can just put `std::vector<value_t> a(source);` here, right? If you're declaring it outside to of the loop to reuse memory, then you can probably just default-initialize it outside of the loop and use `a = source;` here.

Reply:
That would be a bit slower, because the compiler would have to compare the sizes, perform some other unnecessary checks and hopefully call `std::copy` in the end. I mostly used `std::copy` here because that's what the original sorting benchmark does. If you prefer the assignment, I can change that.